public/javascript/posts/BookAppointmentModal.js

```diff
@@ -46,15 +47,23 @@ angular.module('BookAppointmentApp')
     };
 
     $scope.ok = function () {
-      switch($scope.radioAppointment.selected_type){
-        case 1:
-                $scope.addDoctorInfo();
-                break;
-        case 2:
-                $scope.addPatientInfo();
-                break;
+      if($scope.showPatient && $scope.appointmentForm.$invalid){
```

Comment:
I am not able to understand the use case of the two flags -  `$scope.showPatient && $scope.appointmentForm.$invalid`

How are they related ?